src/gsy_e/models/market/future.py

```diff
@@ -178,33 +178,46 @@ def _expire_orders(self, orders: "FutureOrders", current_market_time_slot: DateT
             if time_slot <= current_market_time_slot:
                 del orders.slot_order_mapping[time_slot]
 
-    def delete_orders_in_old_future_markets(self, current_market_time_slot: DateTime) -> None:
+    def delete_orders_in_old_future_markets(self, last_slot_to_be_deleted: DateTime
+                                            ) -> None:
         """Delete order and trade buffers."""
-        self._expire_orders(self.offers, current_market_time_slot)
-        self._expire_orders(self.bids, current_market_time_slot)
+        self._expire_orders(self.offers, last_slot_to_be_deleted)
+        self._expire_orders(self.bids, last_slot_to_be_deleted)
 
         self.offer_history = self._remove_old_orders_from_list(
-            self.offer_history, current_market_time_slot)
+            self.offer_history, last_slot_to_be_deleted)
         self.bid_history = self._remove_old_orders_from_list(
-            self.bid_history, current_market_time_slot)
+            self.bid_history, last_slot_to_be_deleted)
         self.trades = self._remove_old_orders_from_list(
-            self.trades, current_market_time_slot)
+            self.trades, last_slot_to_be_deleted)
 
-    def create_future_markets(self, current_market_time_slot: DateTime,
-                              slot_length: duration,
-                              config: "SimulationConfig") -> None:
-        """Add sub dicts in order dictionaries for future market slots."""
-        if not GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
-            return
-        future_time_slot = current_market_time_slot.add(minutes=slot_length.total_minutes())
-        most_future_slot = (current_market_time_slot +
-                            duration(hours=GlobalConfig.FUTURE_MARKET_DURATION_HOURS))
-        while future_time_slot <= most_future_slot:
+    @staticmethod
+    def _adapt_market_length(market_length: duration, _time_slot: DateTime) -> duration:
+        return market_length
+
+    def _create_future_markets(self,
+                               market_length: duration, start_time: DateTime, end_time: DateTime,
+                               config: "SimulationConfig"):
+        future_time_slot = start_time
+        while future_time_slot <= end_time:
             if (future_time_slot not in self.slot_bid_mapping and
                     is_time_slot_in_simulation_duration(future_time_slot, config)):
                 self.bids.slot_order_mapping[future_time_slot] = []
                 self.offers.slot_order_mapping[future_time_slot] = []
-            future_time_slot = future_time_slot.add(minutes=slot_length.total_minutes())
+            future_time_slot = future_time_slot.add(
+                minutes=self._adapt_market_length(market_length, future_time_slot).total_minutes())
```

Comment:
Since I went here from the leap year calculation, can you explain this a bit more? As far as I see, the time slot duration should be strictly one month and it is not possible by converting the time slot to minutes. Is it really important to have a constant slot length here, and is it guaranteed that this will fail if the slot length changes from one slot to another (e.g. from one month that has 30 days to another one that has 31)?

Reply:
Yes, you are absolutely right, it is not important, that the slot length is constant e.g. for the leapyears.
I need to overwrite the `_adapt_market_length` for the MonthForwardMarket, too, in order to account for the different month length. Thanks for catching this!

Reply:
I removed the `_adapt_market_length` and put the logic in the `_get_market_slot_duration` here f242c73c5b699e83477ef635188acdafda3eb4f0

Also found some things to adapt along the way, e.g. the weekahead market should only start at a Monday.